examples/notebooks/persistent_dataset_speed.ipynb

```diff
@@ -0,0 +1,513 @@
+      "text/plain": [
+       "<Figure size 864x432 with 2 Axes>"
+      ]
+     },
+     "metadata": {
+      "needs_background": "light"
+     },
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "plt.figure('train', (12, 6))\n",
+    "plt.subplot(1, 2, 1)\n",
+    "plt.title('Total Train Time(100 epochs)')\n",
+    "plt.bar('regular', total_time, 1, label='Regular Dataset', color='red')\n",
+    "plt.bar('persistence', persistence_total_time, 1, label='Persistence Dataset', color='blue')\n",
+    "plt.bar('cache', cache_init_time + cache_total_time, 1, label='Cache Dataset', color='green')\n",
+    "plt.bar('cache', cache_init_time, 1, label='Cache Init', color='orange')\n",
+    "plt.ylabel('secs')\n",
+    "plt.grid(alpha=0.4, linestyle=':')\n",
+    "plt.legend(loc='best')\n",
+    "\n",
+    "plt.subplot(1, 2, 2)\n",
+    "plt.title('Epoch Time')\n",
+    "x = [i + 1 for i in range(len(epoch_times))]\n",
+    "plt.xlabel('epoch')\n",
+    "plt.ylabel('secs')\n",
+    "plt.plot(x, epoch_times, label='Regular Dataset', color='red')\n",
+    "plt.plot(x, persistence_epoch_times, label='Persistence Dataset', color='blue')\n",
```

Comment:
Hi @hjmjohnson ,

I got a question here that: why the time consuming of epoch 1 in persistent cache is much faster than regular dataset? I suppose it should be slightly slower because it needs to write cache to files at epoch 1, right?
Thanks.